zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -102,11 +103,13 @@ where
 
     // Height is block height for mined transactions, and the "mempool height" (chain height + 1)
     // for mempool transactions.
-    let height = data
-        .get_tx_height(tx.txid())?
-        .or(data.chain_height()?.map(|max_height| max_height + 1))
-        .or_else(|| params.activation_height(NetworkUpgrade::Sapling))
-        .expect("Sapling activation height must be known.");
+    let height = mined_height.map(Ok).unwrap_or_else(|| {
+        Ok(data
+            .get_tx_height(tx.txid())?
+            .or(data.chain_height()?.map(|max_height| max_height + 1))
+            .or_else(|| params.activation_height(NetworkUpgrade::Sapling))
+            .expect("Sapling activation height must be known."))
+    })?;
```

Comment:
If `mined_height.is_some() && data.get_tx_height(_).is_some()`, and they are not equal, then per the `lightwalletd` data model, `mined_height` is correct, and a reorg has occurred. We therefore would need to signal somewhere that the wallet should proceed with reorg logic.

Reply:
I don't think we need to separately trigger this; the ordinary scanning process should catch the need for the reorg, and having two pathways into triggering that seems more error-prone than having just one.